homeassistant/components/niko_home_control/__init__.py

```diff
@@ -1 +1,31 @@
-"""The niko_home_control component."""
+"""The Niko home control integration."""
+
+from __future__ import annotations
+
+from nikohomecontrol import NikoHomeControlConnection
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PORT, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+PLATFORMS: list[Platform] = [Platform.LIGHT]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set Niko Home Control from a config entry."""
+    controller = NikoHomeControlConnection(entry.data[CONF_HOST], entry.data[CONF_PORT])
+
+    if not controller:
+        raise ConfigEntryNotReady(
+            "cannot connect to controller, please check the host & port are correct."
```

Comment:
```suggestion
            "Cannot connect to controller."
```
If the host and port are incorrect we have more problems tbh, since that should have been prevented in the first place